connectors/storage/v2/api.go

```
@@ -206,8 +214,12 @@ func parseDBError(err error) error { //nolint:funlen // .
 
 			return terror.New(ErrCheckFailed, map[string]any{"column": column})
 		}
-
-		return err
+		if dbErr.SQLState() == "40001" {
+			return terror.New(ErrSerializationFailure, map[string]any{})
```

I would return that error as is:
```
return ErrSerializationFailure
```